src/infill/LightningTreeNode.cpp

```diff
@@ -37,6 +37,23 @@ const Point& LightningTreeNode::getLocation() const
     return p;
 }
 
+Point LightningTreeNode::getFurthestLocation() const
+{
+    Point result{ p };
+    coord_t furthest_dist2 = 0;
+    for (const auto& child : children)
+    {
+        const Point cp{ child->getFurthestLocation() };
+        const coord_t dist2 = vSize2(cp - result);
+        if (dist2 >= furthest_dist2)
+        {
+            furthest_dist2 = dist2;
+            result = cp;
+        }
+    }
```

Comment:
This seems like the sort of stuff that `std::max_element` should be used for?

Reply:
Ah, yes, so you'd leave the `getFurthestLocation` function, but replace the loop and do the recursion in the comparator function argument.